### ISSUE TITLE
Conditional jobs in recipes

### ISSUE THREAD

**Issue by dickmanp** [ngageoint/scale#90]
It would be nice to have some way to conditionally run a job in a recipe.  If I have a job that can produce optional outputs (or an unknown number of output files), I would like to have some way to attach a job to those outputs when they are produced.

**Comment by cshamis** [comment]
We don't want to turn recipes into their own programming language.  With conditions and loops...  But I think that you're onto something there about a parent job (A) that produces (n) output products should have a way to create (n) JOB B children.  The recipe is the same, but the child JOB B might have (n) "clones" of itself, well not clones exactly, but one of itself for every output of (A).  This would be a nice way to "break down" a large problem and let Scale distribute it out.  

I think what we can do now is have Recipe 1: JOB B create a new Scale Trigger Event that launches a Recipe 2: JOB A.  Come to think of it, there'd be nothing stopping it from launching as many as it wanted... but... they'd be permanently in another recipe.  They'd not be able to "come back together" so that might be a consideration.  

Cripes... we're gonna need a Cookbook of Recipe Patterns to help people with some of this stuff.

**Comment by dickmanp** [comment]
That is an interesting idea with the triggers.  However, as you mentioned I would want the results of the "N" jobs to be brought back into the original recipe.

**Comment by dickmancj** [comment]
If we get any more complex with recipes, we should be evaluating existing orchestration standars like BPMN or BPEL. Probably others, but it's been a while since I looked at this problem. There are lots of tools built to handle process definition, execution and tracking.

**Comment by cshamis** [comment]
@cj, agreed.  I don't think we want BPEL, REXX, UML, RPG or bash-n-cron sexiness in the recipes.  We might, read: MIGHT, want to provide for Paul's condition of one job 'cloning' itself wide but that's. IMHO the most complex we want to get.
